test/helpers_test.go

```diff
@@ -266,3 +267,29 @@ func withBuildContext(ctx context.Context, t *testing.T, name string, st llb.Sta
 		sr.FrontendInputs[name] = def.ToPB()
 	}
 }
+
+func reqToState(ctx context.Context, gwc gwclient.Client, sr gwclient.SolveRequest, t *testing.T) llb.State {
+	t.Helper()
+	res := solveT(ctx, t, gwc, sr)
+
+	ref, err := res.SingleRef()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	st, err := ref.ToState()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dt, ok := res.Metadata[exptypes.ExporterPlatformsKey]
+	if ok {
```

Comment:
in what cases would there not be a platform in the metadata?

Reply:
Things like debug/sources wouldn't have a platform set.

Reply:
Oh that's not true, we are using it there... well anything that's not using `frontend.BuildWIthPlatform` would not have a platform set.
Could be some intermediate state that we marshal internally in a test, as an example.